Add "pending" status to ICRC-29

sea-snake opened the pull request:
Add "pending" status to ICRC-29 and clarify expected behavior around it.

sea-snake committed:
Add "pending" status to ICRC-29

Add "pending" status to ICRC-29 and clarify expected behavior around it.

lmuntaner commented on topics/icrc_29_window_post_message_transport.md:
```diff
@@ -41,20 +44,38 @@ The signer should respond to every `icrc29_status` messsage received to indicate
+- `"ready"` indicates the signer is able to receive and process messages immediately.
+- `"pending"` indicates the signer has successfully received the heartbeat and confirms its origin, but is temporarily unable to process further requests.
+
+While `"pending"`, the signer may not respond to subsequent heartbeats for a period of time. Once the signer is able to respond to future requests again, it must resume responding to heartbeats with `"ready"`.
```
> Once the signer is able to respond to future requests again, it must resume responding to heartbeats with `"ready"`

Is this it? Do you mean that after a `"pending"` only a `"ready"` is allowed? I kind of read that from that sentence.

sea-snake replied:
Ah, I'll rephrase. I intended to say:

> The signer must resume responding to heartbeats with "ready" to indicate it is able to respond to future requests again.

lmuntaner replied:
I would change to "future" and instead say something like "next step message requests"

sea-snake replied:
Updated the phrasing to be more clear.

lmuntaner commented on topics/icrc_29_window_post_message_transport.md:
```diff
@@ -41,20 +44,38 @@ The signer should respond to every `icrc29_status` messsage received to indicate
 ## Relying party
 
-> The `origin` value of the first received reply to a `icrc29_status` message with `"result": "ready"` when establishing the communication channel, is mentioned below as `establishedOrigin`.
+> The `origin` value of the first received reply to a `"icrc29_status"` message with `"result": "ready"` when establishing the communication channel, is mentioned below as `establishedOrigin`.
```
Only the `"ready"` or also the `"pending"` is considered an established origin?

sea-snake replied:
Ah yes, this should be either.

lmuntaner commented on topics/icrc_29_window_post_message_transport.md:
````diff
@@ -31,8 +31,11 @@ The message has `targetOrigin` set to `'*'` and is a [JSON-RPC 2.0 (https://www.
     "method": "icrc29_status"
 }
 ```
+The signer should respond to every `"icrc29_status"` message received to indicate its current readiness state. The response message `targetOrigin` should be set to the received `"icrc29_status"` message its `origin` property value.
````
Only the `"ready"` or also the `"pending"` is considered an established origin?

sea-snake replied:
Either (whichever is first) will be considered an established origin. I'll clarify this.

sea-snake committed:
Update conditions for establishedOrigin in documentation

Clarify the conditions for the 'establishedOrigin' value in the communication channel establishment.

sea-snake committed:
Update explanation of 'pending' state behavior

Clarified the behavior of the signer in the 'pending' state regarding heartbeat responses.

lmuntaner commented on topics/icrc_29_window_post_message_transport.md:
```diff
@@ -41,20 +44,38 @@ The signer should respond to every `icrc29_status` messsage received to indicate
 ### Heartbeats
 
-As mentioned above, the relying party continues to send periodic `icrc29_status` messages intended as heartbeat signals, and the signer responds to each received heartbeat with `"result": "ready"`. If the relying party does not receive responses for a given timeframe, it should treat this as a disconnection and stop sending `icrc29_status`
+As mentioned above, the relying party continues to send periodic `"icrc29_status"` messages intended as heartbeat signals, and the signer responds to each received heartbeat with `"result": "ready"`. If the relying party does not receive responses for a given timeframe, it should treat this as a disconnection and stop sending `"icrc29_status"`
```
> the signer responds to each received heartbeat with `"result": "ready"`

Or `"pending"`, right?

lmuntaner replied:
I would put the following in a separate paragraph:

> If the relying party does not receive responses for a given timeframe, it should treat this as a disconnection and stop sending `"icrc29_status"`

sea-snake replied:
Good catch!

sea-snake committed:
Update heartbeat response handling in documentation
